src/core/thread/mle_router.cpp

```diff
@@ -1354,7 +1385,8 @@ otError MleRouter::HandleAdvertisement(const Message &aMessage, const Ip6::Messa
 
         VerifyOrExit(linkMargin >= OPENTHREAD_CONFIG_MLE_PARTITION_MERGE_MARGIN_MIN, error = OT_ERROR_LINK_MARGIN_LOW);
 
-        if ((mDeviceMode & ModeTlv::kModeFFD) &&
+        if (route.IsValid() &&
```

Comment:
This block is to ensure that only newer RouterIdSequence MLE advertisement from previous partition would be handled further, which should apply no matter it is from Router or REED. We might don't want this `route.IsValid()` here?

Reply:
REED does not include Route64 TLV.

Reply:
oh. yes.